### ISSUE TITLE
Provide Dockerfile for thin-edge.io within container

### ISSUE THREAD

**Issue by mbay-SAG** [thin-edge/thin-edge.io#504]
**Is your feature request related to a problem? Please describe.**
Many gateways only provide the possibility to use containers since they hardly relay on standard libraries and do not want them to be changed. There is currently no provided official Dockerfile for running the thin-edge.io within an container.
**Describe the solution you'd like**
A Dockerfile that provided the possibility to build an image for thin-edge.io that directly brings all features and functionalities.
**Describe alternatives you've considered**
Native installation is in several gateways and use cases not possible. Building up an own Dockerfile partly worked since some of the processes can not be automized.

**Comment by marco2901** [comment]
Good Point, #505 states a request to get rid of systemd dependency, which would be also meaningful for this issue as docker containers typically are/should not be equipped with systemd.

**Comment by didier-wenzek** [comment]
There is a first step in that direction https://github.com/thin-edge/thin-edge.io_examples/pull/23 with all the daemons in a single container using alpine and without systemd.

We still have to work so thin-edge can  be deployed over several Docker containers for mosquitto, the mapper and the agent.

However, there is no plan to support these docker images.

**Comment by reubenmiller** [comment]
Relates to #1563

**Comment by reubenmiller** [comment]
The [tedge-demo-container](https://github.com/thin-edge/tedge-demo-container) project has been created to provide a "everything and the kitchensink" approach to getting thin-edge.io running in a systemd environment to simulate running on a real device (without requiring a real device).
